syntax = "proto2";

option cc_enable_arenas = true;

package c2017.shooter;

message ShooterInput {
  required uint64 timestamp = 1;
  required double shooter_encoder_position = 2;
  required double accelarator_encoder_postition = 3;
}

message ShooterOutput {
  required uint64 timestamp = 1;
  required double shooter_voltage = 2;
  required double accelarator_voltage = 3;
}

message ShooterStatus {
  required uint64 timestamp = 1;
  required double observed_velocity = 2; // Positive velocity = forwards [shooter velocity]
  required double accelarator_observed_velocity = 3;
  required double voltage = 4;
  required bool at_goal = 5;
  required bool currently_running = 6;
  required double unprofiled_goal_velocity = 7;
  required double profiled_goal_velocity = 8;
  required double voltage_error = 9;
  required double uncapped_u = 10;
}

enum ShotMode {
  FENDER = 0;
  HOPPER = 1;
}

message ShooterGoal {
  required uint64 timestamp = 1;
  required ShotMode goal_mode = 2;
  required double goal_velocity = 3; // Positive velocity = forwards
}
